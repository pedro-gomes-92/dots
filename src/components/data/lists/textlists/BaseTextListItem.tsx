import React, { FunctionComponent } from 'react';
import { BaseListItemProps, BaseListItem } from '../BaseListItem';
import { TextProps, Text } from '../../../text';
import { ColumnLayout, ColumnItem } from '../../../layouts';

export interface BaseTextListItemProps extends Omit<BaseListItemProps, 'children'> {
  text: TextProps['text'];
  secondaryText?: TextProps['text'];
  size?: TextProps['size'];
}

export const BaseTextListItem: FunctionComponent<BaseTextListItemProps> = ({
  text,
  secondaryText,
  size,
  ...rest
}: BaseTextListItemProps): JSX.Element => (
  <BaseListItem {...rest}>
    <ColumnLayout isMobileActive align="center" gap={2}>
      <ColumnItem>
        <Text className="is-primary" text={text} size={size} />
      </ColumnItem>
      {secondaryText && (
        <ColumnItem size="auto">
          <Text className="is-secondary" text={secondaryText} size={size} />
        </ColumnItem>
      )}
    </ColumnLayout>
  </BaseListItem>
);
