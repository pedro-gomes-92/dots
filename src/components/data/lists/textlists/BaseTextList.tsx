import React, { FunctionComponent } from 'react';

import { Children } from '../../../typings';
import { BaseList, BaseListProps } from '../BaseList';
import { ListDivider } from './ListDivider';
import { BaseTextListItem } from './BaseTextListItem';
import { TextProps } from '../../../text';

export interface BaseTextListProps extends BaseListProps {
  children: Children<typeof BaseTextListItem | typeof ListDivider>;
  size?: TextProps['size'];
}

export const BaseTextList: FunctionComponent<BaseTextListProps> = ({
  children,
  ...rest
}: BaseTextListProps): JSX.Element => <BaseList {...rest}>{children}</BaseList>;
