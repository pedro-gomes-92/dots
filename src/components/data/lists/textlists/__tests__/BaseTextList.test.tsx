import React from 'react';
import { render } from '@testing-library/react';

import { BaseTextList } from '../BaseTextList';
import { BaseTextListItem } from '../BaseTextListItem';

describe('Given <BaseTextList />', (): void => {
  let container: HTMLElement;

  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      container = render(
        <BaseTextList>
          <BaseTextListItem text="Testing" />
        </BaseTextList>,
      ).container;
      expect(container).toMatchSnapshot();
    });
  });
});
