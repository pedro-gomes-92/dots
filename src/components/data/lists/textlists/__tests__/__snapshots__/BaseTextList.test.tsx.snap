// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <BaseTextList /> When initializes Then should match the snapshot 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="list-item"
    >
      <div
        class="layout is-2 columns has-alignment-center is-mobile"
      >
        <div
          class="layout-item column"
        >
          <span
            class="text is-primary"
          >
            Testing
          </span>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
